akka-stream/src/main/scala/akka/stream/impl/Sources.scala

```diff
@@ -136,3 +139,59 @@ private[akka] final class SourceQueueAdapter[T](delegate: SourceQueue[T]) extend
   def offer(elem: T): CompletionStage[QueueOfferResult] = delegate.offer(elem).toJava
   def watchCompletion(): CompletionStage[Done] = delegate.watchCompletion().toJava
 }
+
+/**
+ * INTERNAL API
+ */
+private[stream] final class BlockingStreamSource[T, S](create: () ⇒ S,
```

Comment:
I am not sure about this. Any function T => R can be blocking.

Also, "create" and "close" are not symmetrical, either "open" and "close" or "create" and "destroy".

Isn't this feature more about ReSources? I.e. things which need to be opened and closed?
Is there a reason why a Resource couldn't be non-blocking? as in: create: () => Future[S], read: S => Future[T] and S => Future[Done] ?